src/main/java/wolox/training/models/User.java

```diff
@@ -0,0 +1,58 @@
+package wolox.training.models;
+
+import java.time.LocalDate;
+import java.util.ArrayList;
+import java.util.Collections;
+import java.util.List;
+import javax.persistence.CascadeType;
+import javax.persistence.Entity;
+import javax.persistence.GeneratedValue;
+import javax.persistence.GenerationType;
+import javax.persistence.Id;
+import javax.persistence.ManyToMany;
+import javax.persistence.SequenceGenerator;
+import javax.persistence.Table;
+import javax.validation.constraints.NotNull;
+import lombok.AccessLevel;
+import lombok.Data;
+import lombok.Setter;
+import wolox.training.exceptions.BookAlreadyOwnedException;
+
+@Entity
+@Table(name = "users")
+@Data
+public class User {
+
+    @Id
+    @GeneratedValue(strategy = GenerationType.SEQUENCE, generator = "user_generator")
+    @SequenceGenerator(name = "user_generator", sequenceName = "user_seq")
+    @Setter(AccessLevel.PRIVATE)
+    private Long id;
+
+    @NotNull
+    private String userName;
+
+    @NotNull
+    private String name;
+
+    @NotNull
+    private LocalDate birthDate;
+
+    @ManyToMany(cascade = {CascadeType.PERSIST, CascadeType.MERGE})
+    private List<Book> books = new ArrayList<Book>();
```

Comment:
The cascade types are almost right! We usually use four of this: Persist, Merge, Refresh and Delete
**Persist** says: If you persist the Users instance to the DB, then persist every item that this collection has in the DB as well.
**Delete** says: If you delete the Users instance from the DB, then delete every item that this collection has in the DB as well.
**Merge** says: If you changed the in-memory values of the collection, then merge this changes to the DB so these are reflected there.
**Refresh** says: If something changed in the database values of the collection, then refresh those changes in the in-memory values.

So, MERGE is OK! Because we want to update our database when we save our entities. But PERSIST it's not, because if we save a User then a Book will be try to be saved. And it is possible that the book already exists because we have the POST /api/books endpoint to create them.
Instead of PERSIST, we are missing the one that update our values if the database change :) 
[Here](https://www.baeldung.com/jpa-cascade-types) is a link that explain all of the cascade types if you are interested on that

Reply:
perfect, I will make the change